api/src/task/task.py

```diff
@@ -28,9 +30,18 @@ class Metrics(StrEnum):
     def __init__(self, db_session: db.Session) -> None:
         self.db_session = db_session
         self.metrics: dict[str, Any] = {}
+        self.job: JobTable | None = None
 
     def run(self) -> None:
         try:
+            # Create initial job record
+            job_id = uuid.uuid4()
+            self.job = JobTable(
+                job_id=job_id, job_type=self.cls_name(), job_status=JobStatus.STARTED
+            )
```

Comment:
Should be able to just rely on the db to set it, right?
```suggestion
            self.job = JobTable(
                job_type=self.cls_name(), job_status=JobStatus.STARTED
            )
```